### PR TITLE
ICU-20501 Update test expectations after the changes made for ICU-20273.

### PR DESCRIPTION
**Pull request by roubert**
https://unicode-org.atlassian.net/browse/ICU-20501

### CODE REVIEW

File: icu4c/source/common/locdispnames.cpp [hunk 1]
```diff
@@ -384,6 +384,13 @@ _getDisplayNameForComponent(const char *locale,
 
     root = tag == _kCountries ? U_ICUDATA_REGION : U_ICUDATA_LANG;
 
+    if (displayLocale != NULL && (*displayLocale == '\0' || *displayLocale == '_')) {
+        // Magic string that will have the result that the resource loading code
+        // ends up loading resources for the default locale. (ICU-20501)
+        static const char DEFAULT[] = "default";
```
Comment (srl295) [inline icu4c/source/common/locdispnames.cpp]: make it a constant string (share with `locdspnm.cpp`)

File: icu4c/source/common/locdispnames.cpp [hunk 2]
```diff
@@ -384,6 +384,13 @@ _getDisplayNameForComponent(const char *locale,
 
     root = tag == _kCountries ? U_ICUDATA_REGION : U_ICUDATA_LANG;
 
+    if (displayLocale != NULL && (*displayLocale == '\0' || *displayLocale == '_')) {
```
Comment (markusicu) [inline icu4c/source/common/locdispnames.cpp]: This is not right.
`*displayLocale == '\0'` is the root locale. For that we want to just load "root".
`*displayLocale == '_'` is any locale starting with an empty language subtag, and it makes no sense to just clobber whatever is in there.

File: icu4c/source/common/locdispnames.cpp [hunk 3]
```diff
@@ -384,6 +384,13 @@ _getDisplayNameForComponent(const char *locale,
...
+        // Magic string that will have the result that the resource loading code
```
Comment (markusicu) [inline icu4c/source/common/locdispnames.cpp]: Normally, the displayLocale is passed into ures_open() to load a bundle, and then we look for the right data in that bundle (and its parent chain). The ures_open() implementation should do the right thing, and I remember that one of your recent changes had a fix that I though should handle _IN -> default. We should not need any special handling in this higher-level code.

File: icu4c/source/test/intltest/locnmtst.cpp [hunk 4]
```diff
@@ -422,4 +424,40 @@ void LocaleDisplayNamesTest::TestRootEtc() {
...
+    static const Locale undWithRegion = Locale::forLanguageTag("und-CH", status);
+
+    LocalPointer<LocaleDisplayNames> namesWithoutRegion(LocaleDisplayNames::createInstance(undWithoutRegion));
+    LocalPointer<LocaleDisplayNames> namesWithRegion(LocaleDisplayNames::createInstance(undWithRegion));
+
+    UnicodeString result;
+
+    result.remove();
+    namesWithoutRegion->languageDisplayName("", result);
+    test_assert_equal("Unknown language", result);
```
Comment (markusicu) [inline icu4c/source/test/intltest/locnmtst.cpp]: We should get "root" strings, not English strings, right?

File: icu4c/source/test/intltest/locnmtst.cpp [hunk 5]
```diff
@@ -422,4 +424,40 @@ void LocaleDisplayNamesTest::TestRootEtc() {
...
+    namesWithoutRegion->languageDisplayName("und", result);
+    test_assert_equal("Unknown language", result);
+
+    result.remove();
+    namesWithoutRegion->regionDisplayName("CH", result);
+    test_assert_equal("Switzerland", result);
+
+    result.remove();
+    namesWithRegion->languageDisplayName("", result);
+    test_assert_equal("Unknown language", result);
```
Comment (markusicu) [inline icu4c/source/test/intltest/locnmtst.cpp]: The _CH bundle lookup should go to the default locale (inside ures_open()), so the results here should depend on what the default locale is -- maybe English, maybe not. For testing, we should probably fetch the strings for Locale::getDefault() and use those as expected results.

**Comment by roubert** [comment]
Closing obsolete PR. See PR #576 (ICU-12431) instead for updating these test expectations.

**Pull request closed without merging**